operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -53,26 +59,14 @@ protected Integer currentScale(String namespace, String name) {
      */
     public Future<Void> rollingUpdate(String namespace, String name, long operationTimeoutMs) {
         return getAsync(namespace, name)
-                .compose(deployment -> reconcile(namespace, name, incrementGeneration(deployment)))
+                .compose(deployment -> killPod(namespace, name))
                 .compose(ignored -> readiness(namespace, name, 1_000, operationTimeoutMs));
     }
 
-    private Deployment incrementGeneration(Deployment deployment) {
-        String generationStr = deployment.getSpec().getTemplate().getMetadata().getAnnotations().get(ANNOTATION_GENERATION);
-        int generation = 0;
-        if (generationStr != null && !generationStr.isEmpty()) {
-            generation = Integer.parseInt(generationStr);
-            generation++;
-        }
-        return new DeploymentBuilder(deployment)
-                .editSpec()
-                    .editTemplate()
-                        .editMetadata()
-                            .addToAnnotations(ANNOTATION_GENERATION, Integer.toString(generation))
-                        .endMetadata()
-                    .endTemplate()
-                .endSpec()
-            .build();
+    public Future<ReconcileResult<Pod>> killPod(String namespace, String name) {
```

Comment:
I think `deletePod` better describes what this method does.

Reply:
Agree